### PR TITLE
add a table of contents at the begining of the generated device doc

### LINKED ISSUE #124: fix incorrect jinja comment
## Issue Type

- Documentation Report

## Summary

In the role [eos_l3ls_evpn](https://github.com/aristanetworks/ansible-avd/blob/master/ansible_collections/arista/avd/roles/eos_l3ls_evpn/) , the template [name-servers.j2](https://github.com/aristanetworks/ansible-avd/blob/master/ansible_collections/arista/avd/roles/eos_l3ls_evpn/) use this comment `{# NTP Server #}` instead of `{# name servers #}`

 
## Role or Module Name

role [eos_l3ls_evpn](https://github.com/aristanetworks/ansible-avd/blob/master/ansible_collections/arista/avd/roles/eos_l3ls_evpn/), template [name-servers.j2](https://github.com/aristanetworks/ansible-avd/blob/master/ansible_collections/arista/avd/roles/eos_l3ls_evpn/)

### LINKED ISSUE #282: add a table of content at the begining of the dev documentation outputed by AVD
**Issue Type**

- Documentation template enhancement

**Describe the solution you'd like**

update the template [eos-device-documentation.j2](https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-device-documentation.j2) to include a table of content at the begining 

**Role or Module Name**

eos_cli_config_gen

### LINKED ISSUE #299: template eos-device-documentation.j2 doesnt include some existing templates
## Issue Type

- Documentation Report

## Summary

issue with the template [eos-device-documentation.j2](https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-device-documentation.j2) to include the templates: 
- tacacs-servers.j2
- ip-tacacs-source-interfaces.j2
- aaa-server-groups.j2

here's the issue: 
```
## Spanning Tree

{% include 'documentation/spanning-tree.j2' %}

{# tacacs servers #}

{% include 'documentation/tacacs-servers.j2' %}

{# ip tacacs source interfaces #}

{% include 'documentation/ip-tacacs-source-interfaces.j2' %}

{# aaa server groups #}

{% include 'documentation/aaa-server-groups.j2' %}

## AAA Authentication

{% include 'documentation/aaa-authentication.j2' %}

```

here's the fix: 
```


## Spanning Tree

{% include 'documentation/spanning-tree.j2' %}

## tacacs servers

{% include 'documentation/tacacs-servers.j2' %}

## ip tacacs source interfaces

{% include 'documentation/ip-tacacs-source-interfaces.j2' %}

## aaa server groups

{% include 'documentation/aaa-server-groups.j2' %}

## AAA Authentication

{% include 'documentation/aaa-authentication.j2' %}

```

## Role or Module Name

role `eos_cli_config_gen`

### PR DESCRIPTION
**Pull request by ksator**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

fix #282 

## Component(s) name

`eos_cli_config_gen role`

## Proposed changes

include a table of contents at the begining of the device documentation.  
i.e at the begining of the template [eos-device-documentation.j2](https://github.com/aristanetworks/ansible-avd/blob/devel/ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-device-documentation.j2) 

## How to test

execute the role `eos_cli_config_gen role` and check the devices documentation. 

## Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-device-documentation.j2 [hunk 1]
```diff
@@ -1,5 +1,60 @@
...
+- [Route Maps](#route-maps)  
+- [Peer Filters](#peer-filters)  
+- [Router BFD](#router-bfd)  
+- [Router BGP](#router-bgp)  
+- [Router Multicast](#router-multicast)  
+- [Router PIM Sparse Mode](#router-pim-sparse-mode)  
+- [VM Tracer Sessions](#vm-tracer-sessions)  
+- [Management Security](#management-security)  
+- [Platform](#platform)  
+- [Router ISIS](#router-isis)
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-device-documentation.j2]: Could we move `router isis` just after `router bgp` ?

**Review by titom73 — COMMENTED** [review]
Maybe we should move `router isis` just after `router bgp`

**Comment by aphillipps** [comment]
LGTM

**Review by aphillipps — APPROVED** [review]
LGTM

**Pull request merged**